File: src/com/switkows/mileage/ChartViewer.java
package com.switkows.mileage;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.database.Cursor;
import android.view.View;
import android.view.ViewGroup.LayoutParams;

/**
 * 
 * @author Trevor
 * This is a pretty simple activity: display the specified
 * chart (with correct units), based on the Intent.
 */
public class ChartViewer extends Activity {
   public static final String UNITS_KEY = "units";
   public static final String CHART_KEY = "chartID";
   private Cursor mCursor;
   private Context mContext;
   private MileageChartManager chartManager;

   @Override
   public void onResume() {
      super.onResume();
       mCursor = managedQuery(MileageProvider.CONTENT_URI, null, null, null, null);
       mContext = this;
      chartManager = new MileageChartManager(mContext, mCursor);

       Intent i = getIntent();
       boolean isUS = i.getBooleanExtra(UNITS_KEY, true);
       int chartID = i.getIntExtra(CHART_KEY, MileageChartManager.NO_CHART);
      View view = chartManager.createChart(chartID, isUS);
       addContentView(view, new LayoutParams(LayoutParams.FILL_PARENT,LayoutParams.FILL_PARENT));
   }

}

File: app/src/main/java/com/switkows/mileage/ChartViewer.kt
package com.switkows.mileage

import android.database.Cursor
import android.net.Uri
import android.os.Bundle
import android.preference.PreferenceManager
import android.support.v4.app.FragmentActivity
import android.support.v4.app.LoaderManager
import android.support.v4.content.CursorLoader
import android.support.v4.content.Loader
import android.view.ViewGroup.LayoutParams

/**
 * @author Trevor
 * *         This is a pretty simple activity: display the specified
 * *         chart (with correct units), based on the Intent.
 */
class ChartViewer : FragmentActivity(), LoaderManager.LoaderCallbacks<Cursor> {

    public override fun onResume() {
        super.onResume()
        supportLoaderManager.initLoader(45, null, this)
    }

    override fun onCreateLoader(id: Int, args: Bundle?): Loader<Cursor> {
        val pref = this.getString(R.string.carSelection)
        val car = PreferenceManager.getDefaultSharedPreferences(this).getString(pref, "Car1")
        val uri = Uri.withAppendedPath(MileageProvider.CAR_CONTENT_URI, car)
        return CursorLoader(this, uri, null, null, null, null)
    }

    override fun onLoadFinished(loader: Loader<Cursor>, data: Cursor) {
        val chartManager = MileageChartManager(this, data)

        val i = intent
        val isUS = i.getBooleanExtra(UNITS_KEY, true)
        val chartID = i.getIntExtra(CHART_KEY, MileageChartManager.NO_CHART)
        val view = chartManager.createChart(chartID, true, isUS)
        addContentView(view, LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT))
    }

    override fun onLoaderReset(loader: Loader<Cursor>) {}

    companion object {
        val UNITS_KEY = "units"
        val CHART_KEY = "chartID"
    }

}
